strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/repository/HostedRepositoryProvider.java

```diff
@@ -129,4 +137,25 @@ public Long count(RepositorySearchRequest searchRequest)
                                                    searchRequest.getCoordinates(), searchRequest.isStrict());
     }
 
+    @Override
+    public RepositoryPath getPath(String storageId,
+                                  String repositoryId,
+                                  String path)
+                 throws IOException,
+                        NoSuchAlgorithmException,
+                        ArtifactTransportException,
+                        ProviderImplementationException
+    {
+        Repository repository = getConfiguration().getStorage(storageId).getRepository(repositoryId);
+
+        final LayoutProvider layoutProvider = layoutProviderRegistry.getProvider(repository.getLayout());
+        final RepositoryPath artifactPath = layoutProvider.resolve(repository).resolve(path);
```

Comment:
You can move this line inside the `if` body below, when `if` condition is `true`